src/shogun/machine/Machine.h

```diff
@@ -334,8 +334,35 @@ class CMachine : public CStoppableSGObject
 		 */
 		virtual bool train_machine(CFeatures* data=NULL)
 		{
-			SG_ERROR("train_machine is not yet implemented for %s!\n",
-					get_name());
+			SG_ERROR(
+			    "train_machine is not yet implemented for %s!\n", get_name());
+			return false;
+		}
+
+		virtual bool train_dense(CFeatures* data)
+		{
+			SG_NOTIMPLEMENTED
```

Comment:
Could you be more explicit here: "Training %s with %s is not supported", get_name(), data->get_name()

Reply:
the same error as in CMachine here or ?

Reply:
actually nevermind, this should never be executed anyways